test/non_ssl.jl

```diff
@@ -0,0 +1,302 @@
+    @test !isfile(local_file)
+    body = readstring(server_file)
+    check_response(resp, body, 226, headers)
+end
+
+function test_put(headers, opt)
+    local_file = upload_file
+    server_file = joinpath(ROOT, local_file)
+
+    @test !isfile(server_file)
+    resp = open(local_file) do fp
+        ftp_put(opt, local_file, fp)
+    end
+
+    if isa(opt, ConnContext)
+        ftp_close_connection(opt) # close the connection so that the server file closes
+    end
+
+    @test isfile(server_file)
+    @test readstring(server_file) == readstring(local_file)
+    cleanup_file(server_file)
+
+    check_response(resp, "", 226, headers)
+end
+
+function test_command(headers, opt)
+    resp = ftp_command(opt, "PWD")
+    check_response(resp, "", 257, headers)
+end
+
+function tests_by_mode(active::Bool)
+
+    options = RequestOptions(; opts..., ssl=false, active_mode=active)
+    mode_header = active? "200 Active data connection established." : "229 Entering extended passive mode (...)."
+
+    headers = [
+        "220 pyftpdlib ... ready.",
+        "331 Username ok, send password.",
+        "230 Login successful.",
+        "257 \"/\" is the current directory.",
+        mode_header,
+        "200 Type set to: Binary.",
+        "125 Data connection already open. Transfer starting.",
+        "226 Transfer complete.",
+    ]
+    test_put(headers, options)
+
+    headers = [
+        "220 pyftpdlib ... ready.",
+        "331 Username ok, send password.",
+        "230 Login successful.",
+        "257 \"/\" is the current directory.",
+        mode_header,
+        "200 Type set to: ASCII.",
+        "257 \"/\" is the current directory."
+    ]
+    test_command(headers, options)
+
+
+    headers = [
+        "220 pyftpdlib ... ready.",
+        "331 Username ok, send password.",
+        "230 Login successful.",
+        "257 \"/\" is the current directory.",
+        mode_header,
+        "200 Type set to: Binary.",
+        "213 ...",
+        "125 Data connection already open. Transfer starting.",
+        "226 Transfer complete.",
+    ]
+    test_get(headers, options)
+
+    # download a file to a specific path
+    headers = [
+        "220 pyftpdlib ... ready.",
+        "331 Username ok, send password.",
+        "230 Login successful.",
+        "257 \"/\" is the current directory.",
+        mode_header,
+        "200 Type set to: Binary.",
+        "213 ...",
+        "125 Data connection already open. Transfer starting.",
+        "226 Transfer complete."
+        ]
+
+    save_path = joinpath(pwd(), "test_path.txt")
+    @test !isfile(save_path)
+
+    local_file = download_file
+
+    resp = ftp_get(options, local_file, save_path)
+    server_file = joinpath(ROOT, local_file)
+    body = readstring(server_file)
+
+    @test isfile(save_path) == true
+
+    @test readstring(save_path) == body
+    @test resp.code == 226
+    @test readstring(resp.body) == ""
+    @test is_headers_equal(resp.headers, headers)
```

Comment:
it is missing the test for resp.bytes_recd because it has to account for the save_path as well as the body. I had forgotten to go back and fix it to account for that.